pkg/controller/grafana/grafana_controller.go

```diff
@@ -274,23 +269,33 @@ func (r *ReconcileGrafana) createConfigFiles(cr *i8ly.Grafana) (reconcile.Result
 		return reconcile.Result{}, err
 	}
 
-	for _, resourceName := range []string{common.GrafanaDashboardsConfigMapName, common.GrafanaProvidersConfigMapName, common.GrafanaDatasourcesConfigMapName, common.GrafanaServiceName, ingressType} {
+	resources := []string{
+		common.GrafanaDashboardsConfigMapName,
+		common.GrafanaProvidersConfigMapName,
+		common.GrafanaDatasourcesConfigMapName,
+		common.GrafanaServiceName,
+	}
+	if cr.Spec.Hostname != "" {
```

Comment:
Why only create the Ingress/Route when a Hostname is set? Its not a requirement for OpenShift Routes (gets assigned automatically if not provided) and i believe in vanilla Kubernetes this is valid too.

Reply:
hi @pb82 ... I guess the thinking was to provide a means to control if you wanted a ingress, or defer to a LoadBalancer svc. I was gonna fold this into a spec.ingress.enabled option but wanted to keep it compatible with previous versions. If you prefer though, I'll drop the change and keep the ingress as is?